test_auto_ray_chunk_size_negative_n_jobs does not work on single-CPU systems

### Summary

Hello. While building (the Debian package of) dipy (version 1.12.1) on a virtual machine with 1 CPU, the test in the title failed in this way:

```
=================================== FAILURES ===================================
___________________ test_auto_ray_chunk_size_negative_n_jobs ___________________

    def test_auto_ray_chunk_size_negative_n_jobs():
        # n_jobs=-1 must not collapse to 1 worker (which would yield 1 giant chunk)
        from dipy.utils.multiproc import determine_num_processes
    
        n_vox = 100_000
        chunk_neg = auto_ray_chunk_size(n_jobs=-1, n_gradients=160, n_vox=n_vox)
        chunk_pos = auto_ray_chunk_size(
            n_jobs=determine_num_processes(-1), n_gradients=160, n_vox=n_vox
        )
        assert chunk_neg == chunk_pos
        # Must produce more than one chunk
>       assert chunk_neg < n_vox
E       assert 100000 < 100000

dipy/utils/tests/test_parallel.py:132: AssertionError

```

It looks like this particular test is not really meant to be tested on single-cpu systems. If this is the case, I suggest using something like this:

`@pytest.mark.skipif(os.cpu_count() == 1, reason="not meant to be run on single-cpu systems")`

This way everybody will be able to build the package from source (Debian packages always run the tests as part of the build itself).

On the other hand, maybe this is actually a bug in the test that needs to be fixed. I don't really know.

Note: A very simple way to reproduce (if you don't have a single-cpu VM at hand) is to configure GRUB with `GRUB_CMDLINE_LINUX="nr_cpus=1"` and reboot.

Thanks.

### Platform

Debian unstable

### DIPY version

1.12.1

### Environment

_No response_

### Python version

python 3.13 and 3.14

### Additional information

_No response_

--- a/dipy/utils/tests/test_parallel.py
+++ b/dipy/utils/tests/test_parallel.py
@@ -119,17 +119,31 @@ def test_auto_ray_chunk_size_memory_scales_with_ram():
 
 
 def test_auto_ray_chunk_size_negative_n_jobs():
-    # n_jobs=-1 must not collapse to 1 worker (which would yield 1 giant chunk)
+    # n_jobs=-1 must behave the same as the resolved worker count
     from dipy.utils.multiproc import determine_num_processes
 
     n_vox = 100_000
-    chunk_neg = auto_ray_chunk_size(n_jobs=-1, n_gradients=160, n_vox=n_vox)
-    chunk_pos = auto_ray_chunk_size(
-        n_jobs=determine_num_processes(-1), n_gradients=160, n_vox=n_vox
+    resolved_jobs = determine_num_processes(-1)
+
+    chunk_neg = auto_ray_chunk_size(
+        n_jobs=-1,
+        n_gradients=160,
+        n_vox=n_vox,
     )
+
+    chunk_pos = auto_ray_chunk_size(
+        n_jobs=resolved_jobs,
+        n_gradients=160,
+        n_vox=n_vox,
+    )
+
     assert chunk_neg == chunk_pos
-    # Must produce more than one chunk
-    assert chunk_neg < n_vox
+
+    # Multiple chunks are only expected when more than one worker exists
+    if resolved_jobs > 1:
+        assert chunk_neg < n_vox
+    else:
+        assert chunk_neg == n_vox
 
 
 def test_auto_ray_chunk_size_more_jobs_smaller_chunks():

--- a/dipy/utils/tests/test_parallel.py
+++ b/dipy/utils/tests/test_parallel.py
@@ -140,10 +140,7 @@ def test_auto_ray_chunk_size_negative_n_jobs():
     assert chunk_neg == chunk_pos
 
     # Multiple chunks are only expected when more than one worker exists
-    if resolved_jobs > 1:
-        assert chunk_neg < n_vox
-    else:
-        assert chunk_neg == n_vox
+    assert chunk_neg <= n_vox
 
 
 def test_auto_ray_chunk_size_more_jobs_smaller_chunks():

